bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -0,0 +1,321 @@
+package jsky.app.ot.ags
+
+import java.beans.{PropertyChangeEvent, PropertyChangeListener}
+import java.util.concurrent.{LinkedBlockingQueue, Executors}
+import java.util.logging.Logger
+
+import edu.gemini.ags.api.{AgsRegistrar, AgsStrategy}
+import edu.gemini.pot.sp._
+import edu.gemini.spModel.guide.GuideProbe
+import edu.gemini.spModel.obs.context.ObsContext
+import edu.gemini.spModel.rich.shared.immutable._
+import edu.gemini.spModel.target.env.{TargetEnvironment, GuideProbeTargets}
+import edu.gemini.spModel.target.obsComp.TargetObsComp
+import jsky.app.ot.OT
+import jsky.app.ot.gemini.altair.Altair_WFS_Feature
+import jsky.app.ot.gemini.inst.OIWFS_Feature
+import jsky.app.ot.gemini.tpe.TpePWFSFeature
+import jsky.app.ot.tpe.{GemsGuideStarWorker, GuideStarSupport, TpeContext, TpeManager}
+import scala.collection.JavaConverters._
+import scala.collection.immutable.HashMap
+import scala.swing.Swing
+import scala.util.{Try, Failure, Success}
+import scala.concurrent.ExecutionContext.Implicits.global
+
+
+object BagsManager {
```

Comment:
Can we separate the UI related code from the pure BAGS code? I see in the same class code to handle async AGS calculation with UI updates. It maybe difficult on the OT to separate them though, but having multiple responsibilities makes it harder to follow and test